public/docs/ts/latest/cookbook/graphql.jade

```diff
@@ -0,0 +1,688 @@
+
+.alert.is-important
+  :marked
+    Uri: What is the npm install command for the lodash functions? 
+    Do we install the whole lodash library or just the functions? 
+    (I've never used lodash before).
+    :Kapunahele - fixed. for ease of use I've installed all of lodash
+
+
+:marked
+  Notice that the server includes functions that correspond to each 
+  type in the schema _and_ the mutations.
+
+  This mechanism makes writing simple GraphQL servers straightforward&mdash;you simply 
+  resolve a specific type of data. 
+  This removes the coupling between frontend and backend because if you don't need to know the specific
+  query the client makes to create the server Implementation.
+
+.alert.is-important
+  :marked
+    Uri: I don't know how to incorporate this: "separate the 
+    frontend logic from the backend". And I don't know if 
+    I'm right about "This removes the coupling between frontend and backend".
+    I was trying to make it clearer by breaking it into smaller ideas, but 
+    I'm not sure what those ideas should be.
+    :Kapunahele - expanded on that, let me know if that works
+
+:marked
+  Now, connect the schema to the resolvers with the `makeExecutableSchema` function from
+  the [graphql-tools](http://dev.apollodata.com/tools/graphql-tools/index.html) library:
++makeExample('heroes-graphql/ts/src/app/in-memory-graphql.ts', 'import-graphql-tools', 'in-memory-graphql.ts (excerpt)')
++makeExample('heroes-graphql/ts/src/app/in-memory-graphql.ts', 'make-executable-schema', 'in-memory-graphql.ts (excerpt)')
+
+:marked
+  In the constant `schema`, `makeExecutableSchema` has two properties, 
+  `typeDefs` and `resolvers`. Here, you define them with the `typeDefinitions` 
+  and `resolveFunctions` that you created earlier in `in-memory-graphql.ts`. 
+  This way, your GraphQL server knows where to look for definitions and resolvers.
+
+:marked
+  Now that you have an executable schema, execute it using the `graphql` 
+  library and export it so you can use it with the Apollo Client. 
+  First, `npm install`:
+code-example(language="sh" class="code-shell").
+    npm install graphql --save    
+
+.alert.is-important
+  :marked
+    Uri: can you provide the npm install command?
+    :Kapunahele - done
+
+:marked
+  Next, add an import statement for `execute`.
+
++makeExample('heroes-graphql/ts/src/app/in-memory-graphql.ts', 'import-graphql', 'in-memory-graphql.ts (excerpt)')
+:marked
+  Now create a new `networkInterface` class and call it `InBrowserNetworkInterface`.
+
+  That class will have a `schema` property which it will initialize in the constructor.
+
+  Then create a `query` function that will recieve a query as a variable and execute that query using
+  `graphql` execute function against the schema property.
+
+  You send empty objects to the `rootValue` and `contextValue` arguments of the function and send the 
+  `variables` and `operationName` arguments that are related to the query request.
```

Comment:
This: 
> You send empty objects to the `rootValue` and `contextValue` arguments of the function

Loses me. I get the second part of the sentence about `variables` and `operationName` but where is the first part happening in the code? It may be as simple as saying:

>You send empty objects to the `rootValue` and `contextValue` arguments of the function with X and Y respectively.

X and Y being code snippets like `{}` or whatever is actually doing it.